controlplane/kubeadm/internal/cluster_test.go

```diff
@@ -192,8 +192,10 @@ func TestGetWorkloadCluster(t *testing.T) {
 				}(o)
 			}
 
```

Comment:
The test regularly fails in the second case `returns error if cannot get rest.Config from kubeconfigSecret`.

I think that's because between the first and second case the deletion of the `kubeconfigSecret` is not yet reflected in the local cache.

I think for the purpose of the test it's fine to use the live client.